src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,117 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String SUPPLY_ACTION = "supply";
+    private static final String BUY_ACTION = "buy";
+    private static final String RESULT_ACTION = "result";
+    private static final int INDEX_DATA_OPERATION_TYPE = 0;
+    private static final int INDEX_AMOUNT = 1;
+    private static final int DATA_LENGTH = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        File file = new File(toFileName);
+        try {
+            file.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException("Can't create new file", e);
+        }
+
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file))) {
+            int[] supplies = getSuppliesFromFile(fromFileName);
+            int[] buyies = getBuyiesFromFile(fromFileName);
```

Comment:
first read data, then create report, then start writing data
now you open resource for writing, only then start reading